variables.tf

```diff
@@ -12,6 +12,36 @@ variable "name" {
   description = "The name to give the IBM Event Notification instance created by this module."
 }
 
+variable "cos_destination_name" {
+  type        = string
+  description = "The name to give the IBM Cloud Object Storage destination which will be created for storage of failed delivery events."
+  default     = "COS Destination"
+}
+
+variable "cos_bucket_name" {
+  type        = string
+  description = "The name to use when creating the Cloud Object Storage bucket for storing failed events (NOTE: bucket names are globally unique). If 'add_bucket_name_suffix' is set to true, a random 4 characters will be added to this name to help ensure bucket name is globally unique."
+  default     = null
+}
+
+variable "cos_instance_id" {
+  type        = string
+  description = "The ID of the IBM Cloud Object Storage instance in which the bucket defined in the cos_bucket_name variable exists."
+  default     = null
+}
+
+variable "cos_region" {
```

Comment:
why do you have `cos_region` here? I don't think you are even using it. Also a COS bucket could be cross regional. I don't think you need this variable.